UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs

```diff
@@ -48,5 +98,30 @@ public async Task WhenValidCreateSerialEncFileRequest_ThenReturnTrueResponse()
 
             Assert.AreEqual(true, response);
         }
+
+        #region CreateProductFile
+        [Test]
+        public void WhenInvalidCreateProductFileRequest_ThenReturnFalseResponse()
+        {
+            var salesCatalogueDataResponse = GetSalesCatalogueDataBadrequestResponse();
+
+            var response = fulfilmentAncillaryFiles.CreateProductFile(fakeBatchId, fakeExchangeSetInfoPath, null, salesCatalogueDataResponse);
+
+            Assert.AreEqual(false, response);
+        }
+
+        [Test]
+        public void WhenValidCreateProductFileRequest_ThenReturnTrueResponse()
+        {
+            var salesCatalogueDataResponse = GetSalesCatalogueDataResponse();
+
+            A.CallTo(() => fakeFileSystemHelper.CheckAndCreateFolder(A<string>.Ignored));
+            A.CallTo(() => fakeFileSystemHelper.CreateFileContent(A<string>.Ignored, A<string>.Ignored)).Returns(true);
+
+            var response = fulfilmentAncillaryFiles.CreateProductFile(fakeBatchId, fakeExchangeSetInfoPath, null, salesCatalogueDataResponse);
+
+            Assert.AreEqual(true, response);
```

Comment:
It would be good to add an assertion to ensure that fileSystemHelper.CheckAndCreateFolder is called on the method.

This could be achieved as discussed with Janvi & Shailendra earlier

Reply:
As per suggestion i have done changes.